tools/wavetool.py

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+import os
+import sys
+import argparse
+import numpy as np
+import scipy.signal as signal
+import scipy.io.wavfile as wavefile
+import scipy.fftpack as fftpack
+
+eps = 0.0000000001 # used to avoid zero division error
+
+def generateSinusoid(**params):
+    func = np.sin
+    if params['type'] == 'cosine':
+        func = np.cos
+    time = np.arange(0, float(params['duration']), 1.0/float(params['sample_rate']), dtype=np.float32)
+    data = float(params['amp']) * func(2 * np.pi * float(params['freq']) * time + float(params['phase']))
+    return np.reshape(np.repeat(data, int(params['channel'])),[len(data), int(params['channel'])])
+
+def parse_cmdline():
+    parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+        description='A Tool to Generate and Manipulate Wave Files.')
```

Comment:
If you put a  string at the top of thee file it automatically gets added to the help:
```python
#! /usr/bin/python3
# SPDX-License-Identifier: BSD-3-Clause
# Copyright(c) 2020 Intel Corporation. All rights reserved.

"""
A Tool to Generate and Manipulate Wave Files
"""
```